controllers/operator_controller_test.go

```diff
@@ -112,11 +112,13 @@ var _ = Describe("Reconcile Test", func() {
 					Expect(bd.Spec.Template.Spec.Source.Image.Ref).To(Equal("quay.io/operatorhubio/prometheus@sha256:5b04c49d8d3eff6a338b56ec90bdf491d501fe301c9cdfb740e5bff6769a21ed"))
 				})
 				It("sets resolution success status", func() {
-					cond := apimeta.FindStatusCondition(operator.Status.Conditions, operatorsv1alpha1.TypeReady)
-					Expect(cond).NotTo(BeNil())
-					Expect(cond.Status).To(Equal(metav1.ConditionTrue))
-					Expect(cond.Reason).To(Equal(operatorsv1alpha1.ReasonResolutionSucceeded))
-					Expect(cond.Message).To(Equal("resolution was successful"))
+					Eventually(func() {
```

Comment:
Yes, when the bundleDeployment is first created the conditions do on appear immediately so it moves to `Unknown` state. Removing `Eventually` can cause race conditions wherein the test fails existing at Unknown state.

Reply:
@joelanford  When the first time reconciler results, usually the status of the Operator is `Unknown` because bundle deployment would not have immediately updated its `HasValidBundle` and `InstallSuccessful` status. Eventually doesn't do anything here. I can tweak all other the unit tests to remove conditions checks, and have them in one place (where we verify if bundle status is being translated as right operator status)

Reply:
These tests don't run a controller, so it's up to us to setup the preconditions for each scenario.

Scenario 1: there is no BD, we expect one to be created if resolution succeeds

Scenario 2: there is a BD, but it's spec is wrong. We expect the BD to be updated.

Scenario 3: there is a BD with a specific status. We expect the Operator Ready condition to reflect that status

Other variations of Scenario 3...

Reply:
So I'd expect the condition checks to remain. Just no need for the Eventually wrapper.

Reply:
In each of the scenarios, when the BD is created it initially does not have any conditions set (or it maybe set but Unknown). Which is why the operator returns an Unknown condition, since BD creation is still in progress. As and when the status in BD changes, the corresponding Operator status changes. In the controller, we are checking the BD status as soon as it is created and returning the controller result. The operator status will move from Unknown (Waiting) --> True/False/Unknown. We would not be necessarily getting a true status immediately after the first reconcile because of which the current tests (where BD does not exist before) which check the operator ready condition will fail. 

Agree, eventually wrapper doesn't make sense. But the conditions checks need to be removed.

Reply:
In the scenarios where the BD does not exist or does not have a status before Reconcile starts, we still have deterministic output of the Ready condition state though. We should still be asserting on whatever that output is.